src/backend/ceph.rs

```diff
@@ -651,13 +689,7 @@ impl CephBackend {
         //should ask either another OSD or a monitor.
         let osd_id = get_osd_id_from_device(&self.cluster_handle, dev_path)?;
         debug!("Toggle noscrub, nodeep-scrub flags");
-        osd_set(&self.cluster_handle, &OsdOption::NoScrub, false, simulate)?;
-        osd_set(
-            &self.cluster_handle,
-            &OsdOption::NoDeepScrub,
-            false,
-            simulate,
-        )?;
+        self.set_noscrub(simulate)?;
         debug!("Crush reweight to 0");
         osd_crush_reweight(&self.cluster_handle, osd_id, 0.0, simulate)?;
```

Comment:
Draining an OSD to 0 should also be done incrementally. Setting the weight to 0 would be the same as pulling the drive